### PR TITLE
Feature/DECISION-310

### PR DESCRIPTION
**Pull request by josepablofernandez**

### CODE REVIEW

File: engine/src/main/java/com/stratio/decision/functions/SendToKafkaActionExecutionFunction.java [hunk 1]
```diff
@@ -64,11 +64,15 @@ public Boolean check() throws Exception {
...
-            producer = new Producer<String, String>(new ProducerConfig(properties));
+
+            producer = (Producer) ActionBaseContext.getInstance().getContext().getBean
+                    ("producer");
+
+//            Properties properties = new Properties();
+//            properties.put("serializer.class", "kafka.serializer.StringEncoder");
+//            properties.put("metadata.broker.list", kafkaQuorum);
+//            properties.put("producer.type", "async");
+//            producer = new Producer<String, String>(new ProducerConfig(properties));
```
Comment (compae) [inline engine/src/main/java/com/stratio/decision/functions/SendToKafkaActionExecutionFunction.java]: Remove unnecessary comments??

File: engine/src/main/java/com/stratio/decision/configuration/CassandraConfiguration.java [hunk 2]
```diff
@@ -15,24 +15,52 @@
...
+
+    @Bean
+    public Session cassandraSession() {
+
+        log.debug("Creating Spring Bean for cassandra session");
+        Session session;
+
+        try {
+            session = Cluster.builder().addContactPoints(configurationContext.getCassandraHostsQuorum().split(","))
+                    .withPort(9042)
```
Comment (compae) [inline engine/src/main/java/com/stratio/decision/configuration/CassandraConfiguration.java]: Port in configuration

File: engine/src/main/java/com/stratio/decision/configuration/StreamingContextConfiguration.java [hunk 3]
```diff
@@ -299,23 +321,23 @@ Decision topics has only one partition (by default), so if we have two o more de
...
+//            // if (configurationContext.isAuditEnabled()) {
+//            // SaveRequestsToAuditLogFunction saveRequestsToAuditLogFunction =
+//            // new SaveRequestsToAuditLogFunction(
+//            // configurationContext.getCassandraHostsQuorum());
+//            //
+//            // // persist the RDDs to cassandra using STRATIO DEEP
+//            // allRequests.window(new Duration(2000), new
+//            // Duration(6000)).foreachRDD(saveRequestsToAuditLogFunction);
+//            // }
+//        }
```
Comment (compae) [inline engine/src/main/java/com/stratio/decision/configuration/StreamingContextConfiguration.java]: Remove unnecessary comments??

File: engine/src/main/java/com/stratio/decision/functions/dal/SendToDroolsStreamFunction.java [hunk 4]
```diff
@@ -65,15 +65,21 @@ protected boolean stopAction(StratioStreamingMessage message) {
     @Override
     protected void addStopRequestsValidations(Set<RequestValidation> validators) {
         validators.add(new StreamNameNotEmptyValidation());
-        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
+//        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
```
Comment (danielcsant) [inline engine/src/main/java/com/stratio/decision/functions/dal/SendToDroolsStreamFunction.java]: Remove comments

File: engine/src/main/java/com/stratio/decision/functions/dal/SendToDroolsStreamFunction.java [hunk 5]
```diff
@@ -65,15 +65,21 @@ protected boolean stopAction(StratioStreamingMessage message) {
...
     }
 
     @Override
     protected void addStartRequestsValidations(Set<RequestValidation> validators) {
         validators.add(new StreamNameNotEmptyValidation());
-        validators.add(new ActionEnabledValidation(getStreamOperationService(), StreamAction.SEND_TO_DROOLS,
+//        validators.add(new ActionEnabledValidation(getStreamOperationService(), StreamAction.SEND_TO_DROOLS,
+//                ReplyCode.KO_ACTION_ALREADY_ENABLED.getCode()));
+//        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
+//        validators.add(new DroolsStreamNameValidator());
```
Comment (danielcsant) [inline engine/src/main/java/com/stratio/decision/functions/dal/SendToDroolsStreamFunction.java]: Remove comments

File: engine/src/main/java/com/stratio/decision/functions/ddl/AddQueryToStreamFunction.java [hunk 6]
```diff
@@ -72,15 +72,22 @@ protected boolean stopAction(StratioStreamingMessage message) {
 
     @Override
     protected void addStopRequestsValidations(Set<RequestValidation> validators) {
-        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
-        validators.add(new QueryNotExistsValidation(getStreamOperationService()));
-        validators.add(new UserDefinedStreamValidation(getStreamOperationService()));
+//        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
+//        validators.add(new QueryNotExistsValidation(getStreamOperationService()));
+//        validators.add(new UserDefinedStreamValidation(getStreamOperationService()));
```
Comment (danielcsant) [inline engine/src/main/java/com/stratio/decision/functions/ddl/AddQueryToStreamFunction.java]: Remove comments

File: engine/src/main/java/com/stratio/decision/functions/ddl/AddQueryToStreamFunction.java [hunk 7]
```diff
@@ -72,15 +72,22 @@ protected boolean stopAction(StratioStreamingMessage message) {
...
+        validators.add(new QueryNotExistsValidation());
+        validators.add(new UserDefinedStreamValidation());
     }
 
     @Override
     protected void addStartRequestsValidations(Set<RequestValidation> validators) {
-        validators.add(new QueryExistsValidation(getStreamOperationService()));
-        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
+//        validators.add(new QueryExistsValidation(getStreamOperationService()));
+//        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
```
Comment (danielcsant) [inline engine/src/main/java/com/stratio/decision/functions/ddl/AddQueryToStreamFunction.java]: Remove comments

File: engine/src/main/java/com/stratio/decision/functions/ddl/AlterStreamFunction.java [hunk 8]
```diff
@@ -71,6 +71,7 @@ protected void addStopRequestsValidations(Set<RequestValidation> validators) {
 
     @Override
     protected void addStartRequestsValidations(Set<RequestValidation> validators) {
-        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
+//        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
```
Comment (danielcsant) [inline engine/src/main/java/com/stratio/decision/functions/ddl/AlterStreamFunction.java]: Remove comments

File: engine/src/main/java/com/stratio/decision/functions/ddl/CreateStreamFunction.java [hunk 9]
```diff
@@ -61,14 +61,19 @@ protected boolean stopAction(StratioStreamingMessage message) throws RequestVali
 
     @Override
     protected void addStopRequestsValidations(Set<RequestValidation> validators) {
-        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
-        validators.add(new UserDefinedStreamValidation(getStreamOperationService()));
+//        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
+//        validators.add(new UserDefinedStreamValidation(getStreamOperationService()));
```
Comment (danielcsant) [inline engine/src/main/java/com/stratio/decision/functions/ddl/CreateStreamFunction.java]: Remove comments

File: engine/src/main/java/com/stratio/decision/functions/ddl/CreateStreamFunction.java [hunk 10]
```diff
@@ -61,14 +61,19 @@ protected boolean stopAction(StratioStreamingMessage message) throws RequestVali
...
+        validators.add(new StreamNotExistsValidation());
+        validators.add(new UserDefinedStreamValidation());
     }
 
     @Override
     protected void addStartRequestsValidations(Set<RequestValidation> validators) {
+//        validators.add(new StreamNameNotEmptyValidation());
+//        validators.add(new StreamExistsValidation(getStreamOperationService()));
```
Comment (danielcsant) [inline engine/src/main/java/com/stratio/decision/functions/ddl/CreateStreamFunction.java]: Remove comments

File: engine/src/main/java/com/stratio/decision/functions/dml/InsertIntoStreamFunction.java [hunk 11]
```diff
@@ -73,6 +73,7 @@ protected void addStopRequestsValidations(Set<RequestValidation> validators) {
 
     @Override
     protected void addStartRequestsValidations(Set<RequestValidation> validators) {
-        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
+        //validators.add(new StreamNotExistsValidation(getStreamOperationService()));
```
Comment (danielcsant) [inline engine/src/main/java/com/stratio/decision/functions/dml/InsertIntoStreamFunction.java]: Remove comments

File: engine/src/main/java/com/stratio/decision/service/SaveToCassandraOperationsService.java [hunk 12]
```diff
@@ -26,21 +28,63 @@
...
         this.session = session;
+
+        /*
+        if (session != null) {
+            if (session.getCluster().getMetadata().getKeyspace(STREAMING.STREAMING_KEYSPACE_NAME) == null) {
+                createKeyspace(STREAMING.STREAMING_KEYSPACE_NAME);
+            }
+            refreshTablenames();
+        }
+        */
```
Comment (danielcsant) [inline engine/src/main/java/com/stratio/decision/service/SaveToCassandraOperationsService.java]: Remove comments

File: engine/src/main/java/com/stratio/decision/configuration/StreamingContextConfiguration.java (hunk 3 above)
Comment (danielcsant) [inline engine/src/main/java/com/stratio/decision/configuration/StreamingContextConfiguration.java]: Remove comments

File: engine/src/main/java/com/stratio/decision/functions/dal/SaveToMongoStreamFunction.java [hunk 13]
```diff
@@ -58,15 +58,20 @@ protected boolean stopAction(StratioStreamingMessage message) {
     @Override
     protected void addStopRequestsValidations(Set<RequestValidation> validators) {
         validators.add(new StreamNameNotEmptyValidation());
-        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
+//        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
```
Comment (danielcsant) [inline engine/src/main/java/com/stratio/decision/functions/dal/SaveToMongoStreamFunction.java]: Remove comments

File: engine/src/main/java/com/stratio/decision/functions/dal/SaveToMongoStreamFunction.java [hunk 14]
```diff
@@ -58,15 +58,20 @@ protected boolean stopAction(StratioStreamingMessage message) {
...
     }
 
     @Override
     protected void addStartRequestsValidations(Set<RequestValidation> validators) {
         validators.add(new StreamNameNotEmptyValidation());
-        validators.add(new ActionEnabledValidation(getStreamOperationService(), StreamAction.SAVE_TO_MONGO,
+//        validators.add(new ActionEnabledValidation(getStreamOperationService(), StreamAction.SAVE_TO_MONGO,
+//                ReplyCode.KO_ACTION_ALREADY_ENABLED.getCode()));
+//        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
+//        validators.add(new MongoStreamNameValidator());
```
Comment (danielcsant) [inline engine/src/main/java/com/stratio/decision/functions/dal/SaveToMongoStreamFunction.java]: Remove comments

File: engine/src/main/java/com/stratio/decision/functions/dal/SaveToSolrStreamFunction.java [hunk 15]
```diff
@@ -61,15 +61,20 @@ protected boolean stopAction(StratioStreamingMessage message) {
     @Override
     protected void addStopRequestsValidations(Set<RequestValidation> validators) {
         validators.add(new StreamNameNotEmptyValidation());
-        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
+//        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
```
Comment (danielcsant) [inline engine/src/main/java/com/stratio/decision/functions/dal/SaveToSolrStreamFunction.java]: Remove comments

File: engine/src/main/java/com/stratio/decision/functions/dal/SaveToSolrStreamFunction.java [hunk 16]
```diff
@@ -61,15 +61,20 @@ protected boolean stopAction(StratioStreamingMessage message) {
...
+        validators.add(new StreamNotExistsValidation());
     }
 
     @Override
     protected void addStartRequestsValidations(Set<RequestValidation> validators) {
         validators.add(new StreamNameNotEmptyValidation());
-        validators.add(new ActionEnabledValidation(getStreamOperationService(), StreamAction.SAVE_TO_SOLR,
+//        validators.add(new ActionEnabledValidation(getStreamOperationService(), StreamAction.SAVE_TO_SOLR,
+//                ReplyCode.KO_ACTION_ALREADY_ENABLED.getCode()));
+//        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
```
Comment (danielcsant) [inline engine/src/main/java/com/stratio/decision/functions/dal/SaveToSolrStreamFunction.java]: Remove comments

File: engine/src/test/java/com/stratio/decision/functions/dal/DALActionsFunctionTest.java [hunk 17]
```diff
@@ -41,6 +41,7 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(DALActionsFunctionTest.class);
     private static JavaSparkContext context= null;
 
+
```
Comment (danielcsant) [inline engine/src/test/java/com/stratio/decision/functions/dal/DALActionsFunctionTest.java]: Please remove this changes. Not necessary to commit this class

File: engine/src/test/java/com/stratio/decision/functions/ddl/DDLActionsFunctionTest.java [hunk 18]
```diff
@@ -91,4 +91,5 @@ public void testCreateStreamFunction() throws Exception {
         func.addStartRequestsValidations(validators);
         func.addStopRequestsValidations(validators);
     }
+
```
Comment (danielcsant) [inline engine/src/test/java/com/stratio/decision/functions/ddl/DDLActionsFunctionTest.java]: Please remove this changes. Not necessary to commit this class

File: engine/src/test/java/com/stratio/decision/functions/dml/DMLActionsFunctionTest.java [hunk 19]
```diff
@@ -72,4 +71,5 @@ public void testListStreamFunction() throws Exception {
         func.addStopRequestsValidations(validators);
     }
 
+
```
Comment (danielcsant) [inline engine/src/test/java/com/stratio/decision/functions/dml/DMLActionsFunctionTest.java]: Please remove this changes. Not necessary to commit this class

File: engine/src/test/java/com/stratio/decision/service/SaveToCassandraOperationsServiceTest.java [hunk 20]
```diff
@@ -130,4 +130,5 @@ public void testAlterTable() throws Exception {
         assertEquals("Expected null but exception found", null, ex);
 
     }
+
```
Comment (danielcsant) [inline engine/src/test/java/com/stratio/decision/service/SaveToCassandraOperationsServiceTest.java]: Please remove this changes. Not necessary to commit this class

File: engine/src/test/java/com/stratio/decision/functions/FunctionsTestIT.java [hunk 21]
```diff
@@ -57,7 +58,9 @@ public static void tearDown() throws Exception {
         }
     }
 
+
     @Test
+    @Ignore
```
Comment (danielcsant) [inline engine/src/test/java/com/stratio/decision/functions/FunctionsTestIT.java]: Why this test is ignored?

File: engine/src/test/java/com/stratio/decision/functions/FunctionsTestIT.java [hunk 22]
```diff
@@ -106,6 +110,7 @@ public void testSaveToMongo() throws Exception {
     }
 
     @Test
+    @Ignore
```
Comment (danielcsant) [inline engine/src/test/java/com/stratio/decision/functions/FunctionsTestIT.java]: And this one?

File: engine/src/main/java/com/stratio/decision/functions/PairDataFunction.java [hunk 23]
```diff
@@ -36,6 +36,7 @@
         while (messages.hasNext()) {
             StratioStreamingMessage message = messages.next();
             for (StreamAction action : message.getActiveActions()) {
+
```
Comment (danielcsant) [inline engine/src/main/java/com/stratio/decision/functions/PairDataFunction.java]: Please remove this line. Not necessary to commit this class

File: engine/src/main/java/com/stratio/decision/functions/dal/IndexStreamFunction.java [hunk 24]
```diff
@@ -61,15 +61,20 @@ protected boolean stopAction(StratioStreamingMessage message) {
     @Override
     protected void addStopRequestsValidations(Set<RequestValidation> validators) {
         validators.add(new StreamNameNotEmptyValidation());
-        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
+//        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
```
Comment (compae) [inline engine/src/main/java/com/stratio/decision/functions/dal/IndexStreamFunction.java]: Remove unnecessary comments??

File: engine/src/main/java/com/stratio/decision/functions/dal/IndexStreamFunction.java [hunk 25]
```diff
@@ -61,15 +61,20 @@ protected boolean stopAction(StratioStreamingMessage message) {
...
     }
 
     @Override
     protected void addStartRequestsValidations(Set<RequestValidation> validators) {
         validators.add(new StreamNameNotEmptyValidation());
-        validators.add(new ActionEnabledValidation(getStreamOperationService(), StreamAction.SAVE_TO_ELASTICSEARCH,
+//        validators.add(new ActionEnabledValidation(getStreamOperationService(), StreamAction.SAVE_TO_ELASTICSEARCH,
+//                ReplyCode.KO_ACTION_ALREADY_ENABLED.getCode()));
+//        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
+
```
Comment (compae) [inline engine/src/main/java/com/stratio/decision/functions/dal/IndexStreamFunction.java]: Remove unnecessary comments??

File: engine/src/main/java/com/stratio/decision/functions/dal/ListenStreamFunction.java [hunk 26]
```diff
@@ -57,15 +57,21 @@ protected boolean stopAction(StratioStreamingMessage message) {
     @Override
     protected void addStopRequestsValidations(Set<RequestValidation> validators) {
         validators.add(new StreamNameNotEmptyValidation());
-        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
+//        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
```
Comment (compae) [inline engine/src/main/java/com/stratio/decision/functions/dal/ListenStreamFunction.java]: Remove unnecessary comments??

File: engine/src/main/java/com/stratio/decision/functions/dal/ListenStreamFunction.java [hunk 27]
```diff
@@ -57,15 +57,21 @@ protected boolean stopAction(StratioStreamingMessage message) {
...
 
     @Override
     protected void addStartRequestsValidations(Set<RequestValidation> validators) {
         validators.add(new StreamNameNotEmptyValidation());
-        validators.add(new ActionEnabledValidation(getStreamOperationService(), StreamAction.LISTEN,
+//        validators.add(new ActionEnabledValidation(getStreamOperationService(), StreamAction.LISTEN,
+//                ReplyCode.KO_LISTENER_ALREADY_EXISTS.getCode()));
+//        validators.add(new StreamNotExistsValidation(getStreamOperationService()));
+//        validators.add(new KafkaStreamNameValidator());
+
```
Comment (compae) [inline engine/src/main/java/com/stratio/decision/functions/dal/ListenStreamFunction.java]: Remove unnecessary comments??

File: engine/src/main/java/com/stratio/decision/service/SaveToCassandraOperationsService.java [hunk 28]
```diff
@@ -26,21 +28,63 @@
...
 
-    private final Session session;
+    private  final Session session;
+
+    private HashMap<String, Integer> tablenames = new HashMap<>();
 
     public SaveToCassandraOperationsService(Session session) {
+
         this.session = session;
+
```
Comment (compae) [inline engine/src/main/java/com/stratio/decision/service/SaveToCassandraOperationsService.java]: Remove unnecessary comments??

File: engine/src/test/java/com/stratio/decision/functions/FunctionsTestIT.java (hunk 22 above)
Comment (compae) [inline engine/src/test/java/com/stratio/decision/functions/FunctionsTestIT.java]: ignore?

File: engine/src/test/java/com/stratio/decision/functions/FunctionsTestIT.java (hunk 21 above)
Comment (compae) [inline engine/src/test/java/com/stratio/decision/functions/FunctionsTestIT.java]: ignore?

File: engine/src/main/java/com/stratio/decision/configuration/CassandraConfiguration.java (hunk 2 above)
Comment (josepablofernandez) [inline engine/src/main/java/com/stratio/decision/configuration/CassandraConfiguration.java]: this issue will be resolved merging DECISION-217 PR

**Comment by compae** [comment]
Make squash!! only one commit

**Comment by danielcsant** [comment]
+1!

**Comment by dvallejo** [comment]
+1

**Comment by gbecares** [comment]
Cassandra bug solved. +1

**Comment by gbecares** [comment]
:shipit:

**Comment by stratiocommit** [comment]
Comment accepted. Will treat it ASAP.

**Pull request merged**